### PR TITLE
Add Runtime Benchmarking

### PR DESCRIPTION
**Pull request by shawntabrizi**
This PR adds the FRAME benchmarking pipeline to the Kulupu node.

You can now compile this project with the `--features runtime-benchmarks` flag, and it will generate a node binary that you can use to execute runtime benchmarks. Otherwise, the node binary generation should be unaffected.

To run a benchmark, just run:

```bash
kulupu benchmark --pallet <name> --extrinisic <name> --other-options
```

Here are all the available flags:

```bash
➜  kulupu git:(shawntabrizi-add-benchmarking) ./target/release/kulupu benchmark --help                           
kulupu-benchmark 2.0.0
Benchmark runtime pallets.

USAGE:
    kulupu benchmark [FLAGS] [OPTIONS] --extrinsic <extrinsic> --pallet <pallet>

FLAGS:
        --dev                 
            Specify the development chain

        --extra               
            Display and run extra benchmarks that would otherwise not be needed for weight construction

    -h, --help                
            Prints help information

        --no-median-slopes    
            Don't print the median-slopes linear regression analysis

        --no-min-squares      
            Don't print the min-squares linear regression analysis

        --output              
            Output the benchmarks to a Rust file

        --raw                 
            Print the raw results

    -V, --version             
            Prints version information

        --weight-trait        
            Output the trait definition to a Rust file


OPTIONS:
    -d, --base-path <PATH>                  
            Specify custom base path

        --chain <CHAIN_SPEC>                
            Specify the chain specification (one of dev, local, or staging)

        --db-cache <MiB>                    
            Limit the memory the database cache can use [default: 128]

        --execution <STRATEGY>              
            The execution strategy that should be used for benchmarks [possible values: Native, Wasm, Both,
            NativeElseWasm]
    -e, --extrinsic <extrinsic>             
            Select an extrinsic inside the pallet to benchmark, or `*` for all

        --heap-pages <heap-pages>           
            Set the heap pages while running benchmarks

        --high <highest-range-values>...    
            Indicates highest values for each of the component ranges

    -l, --log <LOG_PATTERN>...              
            Sets a custom logging filter. Syntax is <target>=<level>, e.g. -lsync=debug.
            
            Log levels (least to most verbose) are error, warn, info, debug, and trace. By default, all targets log
            `info`. The global log level can be set with -l<level>.
        --low <lowest-range-values>...      
            Indicates lowest values for each of the component ranges

    -p, --pallet <pallet>                   
            Select a FRAME Pallet to benchmark, or `*` for all (in which case `extrinsic` must be `*`)

    -r, --repeat <repeat>                   
            Select how many repetitions of this benchmark should run [default: 1]

    -s, --steps <steps>...                  
            Select how many samples we should take across the variable components

        --wasm-execution <METHOD>           
            Method for executing Wasm runtime code [default: Interpreted]  [possible values: Interpreted]
```

### CODE REVIEW

**Comment by shawntabrizi** [comment]
This is ready for review.

File: Cargo.toml [hunk 1]
```diff
@@ -60,6 +64,9 @@ wasmtime = [
 	"sc-cli/wasmtime",
 	"sc-service/wasmtime",
 ]
+runtime-benchmarks=[
```
Comment (sorpaas) [inline Cargo.toml:67]:
```suggestion
runtime-benchmarks = [
```

File: src/command.rs [hunk 2]
```diff
@@ -154,6 +157,17 @@ pub fn run() -> sc_cli::Result<()> {
 				Ok(())
 			})
 		},
+		Some(Subcommand::Benchmark(cmd)) => {
+			if cfg!(feature = "runtime-benchmarks") {
+				let runner = cli.create_runner(cmd)?;
+
+				runner.sync_run(|config| cmd.run::<kulupu_runtime::Block, service::Executor>(config))
+			} else {
+				println!("Benchmarking wasn't enabled when building the node. \
```
Comment (sorpaas) [inline src/command.rs:166]: Perhaps return an error here?

```suggestion
				Err("Benchmarking wasn't enabled when building the node. \
				     You can enable it with `--features runtime-benchmarks`.to_string())
```

**Review by sorpaas — APPROVED** [review]

**Comment by shawntabrizi** [comment]
Next steps (in a follow up PR) are to write benchmarks for the additional pallets introduced, and collect weights

**Pull request merged**